target/linux/realtek/files-5.10/drivers/net/dsa/rtl83xx/rtl838x.h

```diff
@@ -551,6 +551,14 @@ typedef enum {
 #define RTL930X_L3_IP_MTU_CTRL(i)		(0xAB5C + ((i >> 1) << 2))
 #define RTL930X_L3_IP6_MTU_CTRL(i)		(0xAB6C + ((i >> 1) << 2))
 #define RTL930X_L3_HW_LU_KEY_CTRL		(0xAC9C)
+#define RTL930X_L3_HW_LU_KEY_CTRL_ROUTE_TYPE_MASK       GENMASK(20, 19)
+#define RTL930X_L3_HW_LU_KEY_CTRL_ROUTE_TYPE_OFFSET     19
+#define RTL903X_L3_HW_LU_KEY_CTRL_ROUTE_TYPE(r) \
+        (((r) << RTL930X_L3_HW_LU_KEY_CTRL_ROUTE_TYPE_OFFSET) & RTL930X_L3_HW_LU_KEY_CTRL_ROUTE_TYPE_MASK)
```

Comment:
To make this more compact, you could use `FIELD_SET()` instead. Then you don't need the _OFFSET macro any more.

Reply:
From bitfield.h, but apparently my memory failed me and it's called FIELD_PREP()
https://elixir.bootlin.com/linux/v6.1-rc2/source/include/linux/bitfield.h#L126